### PR TITLE
Implemented token based authentication for django rest framework

### PR DESCRIPTION
**Pull request by shreygoel7**

### CODE REVIEW

**Review by dynamoh — CHANGES_REQUESTED** [review]
REST API views and HttpResponse views should be kept separate.
Folder structure should be :
```
- globals
    - api
         - views.py
         - urls.py
         - serializers.py
```

**Comment by garg3133** [comment]
@shreygoel7 Can you please remove all the migrations from this PR (and also delete them from your system). I've fixed all the migration issues on Fusion and merged them in #280.
(We need to take extra care regarding the migrations otherwise it may cause problems in the database. We'll soon release the guidelines for making migrations).

Steps to be followed:
* Delete all the 7 migration files you created and pushed in this PR.
* Stage them (`git add <name of 7 files>` or use Git tab in vscode to do this using GUI).
* Run `git commit --amend`. (It will edit the previous commit you made so make sure you didn't commit anything after making this PR).
* Now force push your modified commit to this PR (`git push origin token-based-auth`).

Also, please don't run migrations command on your system. Whenever you make any changes in the models and want to migrate them, commit the changed models files and create a PR and I will migrate those changes for you (it will ensure consistency among migrations and will keep the database safe).

Let me know if you face any problem :)

File: FusionIIIT/Fusion/settings/development.py [hunk 1]
```diff
@@ -16,6 +16,15 @@
     }
 }
 
+REST_FRAMEWORK = {
```
Comment (garg3133) [inline FusionIIIT/Fusion/settings/development.py:19]: Doesn't this needs to be added in productions settings also?

**Review by garg3133 — COMMENTED** [review]
Also as @dynamoh said, you should write the views and urls for API in separate files.

views should be written in `globals/api/views.py` and urls in `globals/api/urls.py` files.

**Comment by shreygoel7** [comment]
Made all the changes

File: FusionIIIT/Fusion/settings/development.py (hunk 1 above)
Comment (shreygoel7) [reply FusionIIIT/Fusion/settings/development.py:19]: Added in productions settings.

**Review by dynamoh — CHANGES_REQUESTED** [review]
- [x] change base route from ```/api/``` to ```/api/auth/``` for auth APIs.
- [x] Instead of GenericViewset create an APIView for Login.

**Review by dynamoh — APPROVED** [review]

**Pull request merged**